cwltool/expression.py

```diff
@@ -135,13 +169,19 @@ def do_eval(ex, jobinput, requirements, outdir, tmpdir, resources,
         "self": context,
         "runtime": runtime }
 
-    if isinstance(ex, dict) and "engine" in ex and "script" in ex:
-        return exeval(ex, jobinput, requirements, outdir, tmpdir, context, pull_image)
     if isinstance(ex, (str, unicode)):
+        fullJS = False
+        jslib = ""
```

Comment:
Perhaps try jslib=u"" or jslib=None to make your incompatible type check go away.